integration_test/ops_agent_test.go

```diff
@@ -2388,9 +2388,6 @@ func TestResourceDetectorOnGCE(t *testing.T) {
 	t.Parallel()
```

Comment:
Maybe not in scope for this PR, but why do we even need this test to begin with? Is it not feasible to test the resource detector via a prometheus pipeline? It might be simpler that way; there's quite a bit of machinery here for this test.

It's also important to call out that the version of `resourcedetector` under test here does not necessarily match the version under test in the rest of the integration tests -- those tests are done against the installed Ops Agent package (which could be the latest public release, could be from REPO_SUFFIX, or could be from AGENT_PACKAGE_IN_GCS), while this test is done against HEAD. Not much of an issue for the automated test pipelines but this could affect anyone testing locally.

If it's not feasible to test `resourcedetector` via prometheus, there are some ways we can still simplify this:

* We can just clone the Ops Agent repo on the test VM instead of needing to specify each individual file to upload
* If that's not preferable, then you can just upload the whole `cmd/run_resource_detector` directory and generate `go.mod` automatically -- then we only have to define one thing to upload instead of six
* The regexes don't add value IMO and can be removed
* We don't need to set up the Ops Agent for this test, we just need a VM
* The assertions can each be condensed into e.g. `assert.Equal(t, actual.InstanceName, vm.Name)` (you can import the `assert` package that we use in `confgenerator_test`)

Reply:
Those are good points! I think I will have a separate PR to address those suggestions. 
- Why this test is needed: we have another test `TestPrometheusMetrics` which checks the detected metadata via a Prometheus pipeline. it covers many similar checks as this `resourcedetector` test. This standalone `resourcedetector` was added because: 1. The Prometheus receiver would transform the metadata to the Prometheus format, so with this test we can check the raw values, as a way to make sure the data source (metadata server on GCE) always returns expected values. 2. The `resourcedetector` and some of its fields might be used by other components other than the Prometheus receiver. Though we probably can find ways to cover those two without this test. Let me know what you think. 
- Regarding the issue that this test always test against local HEAD: this is a good point - indeed this is a different behaviour here. My thought was by doing it this way, we can test local changes when running this test locally. The options of cloning the entire repo from a branch or generating `go.mod` and resolving the package automatically would not allow us to test local changes. Again let me know what you think here. I should at least add comments to the test to highlight that this always tests against HEAD and ignore the provided Ops Agent. 
- Yeah I think the regex for extract the json from stdout is not needed. Will remove. 
- Yes - we do not need to set up the Ops Agent for this test. Do you mean I can skip `agents.CommonSetup(t, platform)` and instead just create a VM? 
- Good suggestion - I will replace those with `assert.Equal`!

Again thanks for the questions! I could have this PR ready first to unblock the nightlies and have a separate PR to improve this test. Or I can revert the previous PR to remove this test for now from master while working on those.